enhancements/api-review/add-repositoryMirrors-spec.md

```diff
@@ -0,0 +1,189 @@
+---
+title: add-repositoryMirrors-spec
+authors:
+  - "@QiWang19"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2021-03-10
+last-updated: 2021-03-10
+status: implementable
+---
+
+# Add repositoryMirrors spec to cluster wide ImageContentSourcePolicy
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Today, the ImageContentSourcePolicy object sets up mirror with `mirror-by-digest-only` property set to true. Requires images be pulled by digest only. This enhancement plans to add new spec to ImageContentSourcePolicy with configurable field `allowMirrorByTags`, so that mirror-by-digest-only` can be configured by ImageContentSourcePolicy.
+
+## Motivation
+
+Today, the ImageContentSourcePolicy object sets up mirror configuration with `mirror-by-digest-only` property set to true, which leads to using mirrors only when images are referenced by digests. However, when working with disconnected environments, sometimes multiple ImageContentSourcePolicies are needed, some of them are used by apps/manifests that don't use digests when pulling the images.
+
+Adding the `repositoryMirrors` spec to ImageContentSourcePolicy to configure mirrors will make it possible to pull images from mirror using tags without requiring image digest reference.
+
+### Goals
+
+- Enable the user to pull images through ImageContentSourcePolicy configured mirrors by tags.
+
+### Non-Goals
+
+- The `allowMirrorByTags` will not be configurable or applied to OpenShift payload images.
+
+## Proposal
+
+The new spec `repositoryMirrors` will be added to ImageContentSourcePolicy. The `allowMirrorByTags` will be the configurable filed can allow using mirrors without requiring digest reference.
+
+```go
+// RepositoryMirrors holds cluster-wide information about how to handle mirrors in the registries config.
+// Note: this is different from the RepositoryDigestMirrors that the mirrors only 
+// work when pulling the images that are referenced by their digests.
+type RepositoryMirrors struct {
+	// source is the repository that users refer to, e.g. in image pull specifications.
+	// +required
+	Source string `json:"source"`
+  // If true, mirrors will only be used for digest pulls. Pulling images by
```

Comment:
nit: inconsistent indenting flipping between tabs and spaces.

Reply:
Fixed.